connector/src/main/java/org/geysermc/connector/entity/FishingHookEntity.java

```diff
@@ -26,43 +26,160 @@
 package org.geysermc.connector.entity;
 
 import com.github.steveice10.mc.protocol.data.game.entity.metadata.EntityMetadata;
-import com.github.steveice10.mc.protocol.data.game.entity.object.ProjectileData;
 import com.nukkitx.math.vector.Vector3f;
+import com.nukkitx.math.vector.Vector3i;
 import com.nukkitx.protocol.bedrock.data.entity.EntityData;
-import org.geysermc.connector.GeyserConnector;
+import com.nukkitx.protocol.bedrock.data.entity.EntityFlag;
+import com.nukkitx.protocol.bedrock.packet.PlaySoundPacket;
+import org.geysermc.connector.entity.player.PlayerEntity;
 import org.geysermc.connector.entity.type.EntityType;
 import org.geysermc.connector.network.session.GeyserSession;
+import org.geysermc.connector.network.translators.collision.BoundingBox;
+import org.geysermc.connector.network.translators.collision.CollisionManager;
+import org.geysermc.connector.network.translators.collision.CollisionTranslator;
+import org.geysermc.connector.network.translators.collision.translators.BlockCollision;
+import org.geysermc.connector.network.translators.world.block.BlockStateValues;
+import org.geysermc.connector.network.translators.world.block.BlockTranslator;
 
-public class FishingHookEntity extends Entity {
-    public FishingHookEntity(long entityId, long geyserId, EntityType entityType, Vector3f position, Vector3f motion, Vector3f rotation, ProjectileData data) {
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class FishingHookEntity extends ThrowableEntity {
+
+    private boolean hooked = false;
+
+    private final BoundingBox boundingBox;
+
+    private boolean inWater = false;
+
+    public FishingHookEntity(long entityId, long geyserId, EntityType entityType, Vector3f position, Vector3f motion, Vector3f rotation, PlayerEntity owner) {
         super(entityId, geyserId, entityType, position, motion, rotation);
 
-        for (GeyserSession session : GeyserConnector.getInstance().getPlayers()) {
-            Entity entity = session.getEntityCache().getEntityByJavaId(data.getOwnerId());
-            if (entity == null && session.getPlayerEntity().getEntityId() == data.getOwnerId()) {
-                entity = session.getPlayerEntity();
-            }
+        this.boundingBox = new BoundingBox(0.125, 0.125, 0.125, 0.25, 0.25, 0.25);
 
-            if (entity != null) {
-                this.metadata.put(EntityData.OWNER_EID, entity.getGeyserId());
-                return;
-            }
-        }
+        // Silence splash sounds from bedrock
```

Comment:
Why do we want to silence splash sounds from Bedrock?

Reply:
In the first video of https://github.com/GeyserMC/Geyser/issues/1780, the sound is due to the hook coming out of the water and going back in. There is no difference between the sound from a fish being caught and this splashing sound as its volume doesn't depend on the hook's velocity. I've silenced this sound to allow the code in `moveAbsoluteImmediate` to handle the splash similar to Java. I'm pretty sure Bedrock deals with this by keeping the hook under the water.

Reply:
Gotcha. A comment here in the code explaining this would be welcome - feel free to go into the same extensive detail.